server/svix-server/src/v1/endpoints/endpoint/mod.rs

```diff
@@ -474,12 +474,45 @@ impl ModelIn for EndpointHeadersPatchIn {
     }
 }
 
+#[derive(Deserialize, JsonSchema)]
+struct EndpointStatsRange {
+    since: Option<DateTime<Utc>>,
+    until: Option<DateTime<Utc>>,
+}
+
+impl EndpointStatsRange {
+    fn validate_unwrap_or_default(self) -> error::Result<(DateTime<Utc>, DateTime<Utc>)> {
+        let until = self.until.unwrap_or_else(Utc::now);
+
+        if until > Utc::now() {
+            return Err(HttpError::bad_request(
+                Some("invalid_range".into()),
+                Some("'until' cannot be in the future".into()),
+            )
+            .into());
+        }
+
+        let since = self.since.unwrap_or(until - Duration::days(28));
+        if until - since > Duration::days(28) {
```

Comment:
I would probably play extra-safe with this and add a bit of an epsilon to the allowed window in the condition here, because this can be weird due to leap seconds or other calendar weirdness.